deployer/utils.py

```diff
@@ -130,6 +135,21 @@ def verify_and_decrypt_file(encrypted_path):
         return
 
 
+@contextmanager
+def get_decrypted_files(files, abspath):
+    """
+    This is a context manager that when entered provides a list of
+    file paths, where temporary files have been created if needed for
+    files that were encrypted and first need to be decrypted.
+    """
+    with ExitStack() as stack:
+        yield [
+            stack.enter_context(verify_and_decrypt_file(abspath.joinpath(f)))
+            for f in files
+            if check_file_exists(abspath.joinpath(f))  # Check the file exists
```

Comment:
Hmmm, this is always true or error - so no filtering is done. You are not really checking you are asserting.

What do you think about renaming the function to be "assert" file exist maybe? That would read more clear to me because I read this as you may end up filter out files that doesn't exist which to me should instead raise an error.

I also wonder, if we have a function named "verify_and_decrypt_file" - it becomes very surprising if that would just cleanly return a file path that didn't exist - but only verify that encrypted files wasn't decrypted already.

I considered if "verify" should be included in the function name - it is describing what it does in a way, but fail to do so in clear enough detail to really help me understand. If we think it is helpful, then in a way, `get_decrypted_files` should perhaps be named similarly, because it also verifies files in a way after all - but it relies on the helper functions - both for encrypted/not-encrypted files.

Concrete ideas:
1. Rename `check_file_exists` to `assert_file_exists`.
2. Let `assert_file_exists` run for the `verify_and_decrypt_file` function to assert a file to exist no matter what as its first verification check.

Reply:
Done in most recent commits